src/indicator.js

```diff
@@ -58,22 +58,25 @@ var CPUFreqIndicator = class CPUFreqIndicator extends baseindicator.CPUFreqBaseI
         this.cpufreq = 800;
         this.cpucount = 0;
         this.isTurboBoostActive = true;
+        this.isAutoSwitchActive = false;
```

Comment:
I think the AutoSwitch feature should always be active, as setting the `Default Battery Profile` in the settings without the extension switching to this profile might confuse users. A better UX would be to always switch to the default profile and have a `Do Nothing` option in the default profile selection.

Reply:
My intention for the switch in the popup menu was that you can pause automatic switching if you want to stay in another profile for a while regardless of the power state.

Reply:
Hmm, I'm not sure whether this is needed. If a user wants to stay on a `Multimedia Profile`, switching the profile back manually after the auto-switching is done, or disabling auto-switch, leads to the same amount of steps a user has to take. With the difference that the auto-switch feature stays disabled and the manual profile switch will be overridden with the next power event. I'm not sure what the best approach would be...

@martin31821 any thoughts?

Reply:
The idea seems good, I'd keep it in there. 
Is that switch persistent or only in memory?

Reply:
The switch state is stored in `LASTSETTINGS`

Reply:
I would suggest setting `isAutoSwitchActive` to `true` as a default.